docs/ROLLOUT_STRATEGY_GUIDE.md

```diff
@@ -0,0 +1,309 @@
+
+## Rollout Modes
+
+### Availability Mode
+- **Purpose**: Ensures high availability during deployments
+- **Behavior**: Launches new pods first, then terminates old pods
+- **Kubernetes Settings**: `maxUnavailable=0`, `maxSurge=<ratio>`
+- **Use Case**: Production environments where downtime is not acceptable
+
+### ResourceAware Mode
+- **Purpose**: Optimizes resource usage during deployments
+- **Behavior**: Terminates old pods first, then launches new pods
+- **Kubernetes Settings**: `maxSurge=0`, `maxUnavailable=<ratio>`
+- **Use Case**: Resource-constrained environments or cost optimization
+
+## Configuration
+
+### InferenceService Level Configuration
+
+You can configure rollout strategy at the component level (predictor, transformer, explainer) within your InferenceService:
+
+```yaml
+apiVersion: serving.kserve.io/v1beta1
+kind: InferenceService
+metadata:
+  name: rollout-strategy-example
+  namespace: default
+  annotations:
+    serving.kserve.io/deploymentMode: "RawDeployment"
+spec:
+  predictor:
+    model:
+      modelFormat:
+        name: sklearn
+      storageUri: "s3://my-bucket/model"
+    # Availability mode - high availability
+    rollout:
+      mode: "Availability"
+      ratio: "50%"
+    
+  transformer:
+    custom:
+      container:
+        image: my-transformer:latest
+    # ResourceAware mode - resource efficiency
+    rollout:
+      mode: "ResourceAware"
+      ratio: "25%"
+```
+
+### Configuration Parameters
+
+- **mode**: Either `"Availability"` or `"ResourceAware"`
+- **ratio**: Can be specified as:
+  - Percentage (e.g., `"25%"`, `"50%"`)
+  - Absolute number (e.g., `"2"`, `"5"`)
+
+### Default Configuration
+
+### ConfigMap Defaults
+
+If no rollout strategy is specified in the InferenceService, KServe will use default values from the ConfigMap:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: inferenceservice-config
+  namespace: kserve
+data:
+  deploy: |-
+    {
+      "defaultDeploymentMode": "Serverless",
+      "rawDeploymentRolloutStrategy": {
+        "defaultRollout": {
+          "mode": "Availability",
+          "ratio": "25%"
+        }
+      }
+    }
+```
+
+### KServe Defaults (When No ConfigMap Defaults)
+
+If neither the InferenceService spec nor the ConfigMap defines rollout strategy values, KServe applies its own defaults:
+
+- **maxUnavailable**: `25%`
+- **maxSurge**: `25%`
+
+### Special Case: Multinode Deployments
+
+For multinode deployments, KServe automatically overrides the rollout strategy to ensure high availability:
+
+- **maxUnavailable**: `0%` (no pods are taken down during rollout)
+- **maxSurge**: `100%` (can have up to double the number of pods during rollout)
+
+This ensures that original pods remain available until new pods are ready.
```

Comment:
@Jooho can you help review if this is ok?

Reply:
https://github.com/kserve/kserve/blob/master/pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go#L196 This is the logic of the code for multi-node. I just updated based the logic here.